mlrun/api/crud/model_endpoints.py

```diff
@@ -59,20 +67,29 @@ def create_or_patch(
                 )
             )
 
+            # get stats from model object if not found in model endpoint object
             if not model_endpoint.status.feature_stats and hasattr(
                 model_obj, "feature_stats"
             ):
                 model_endpoint.status.feature_stats = model_obj.feature_stats
 
+            # get labels from model object if not found in model endpoint object
             if not model_endpoint.spec.label_names and hasattr(model_obj, "outputs"):
                 model_label_names = [
                     self._clean_feature_name(f.name) for f in model_obj.outputs
                 ]
                 model_endpoint.spec.label_names = model_label_names
 
+            # get algorithm from model object if not found in model endpoint object
             if not model_endpoint.spec.algorithm and hasattr(model_obj, "algorithm"):
                 model_endpoint.spec.algorithm = model_obj.algorithm
 
+            # Create monitoring feature set if monitoring found in model endpoint object
+            if model_endpoint.spec.monitoring_mode == "enabled":
+                self.create_monitoring_feature_set(
+                    model_endpoint, model_obj, db_session
+                )
```

Comment:
Lets move it to struct that contains the different modes, so we could refer to the available modes.